src/components/NameServerGroupUpdate.tsx

```diff
@@ -254,7 +254,7 @@ const NameServerGroupUpdate = () => {
   };
 
   const domainRegex =
-    /(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]/;
+    /^(?!.*\s)[a-zA-Z0-9](?!.*\s$)(?!.*\.$)(?:(?!-)[a-zA-Z0-9-]{1,63}(?<!-)\.){1,126}(?!-)[a-zA-Z0-9-]{1,63}(?<!-)$/;
```

Comment:
looks fine but can you move this to the utils path so we keep one domainRegex logic and domainValidator ?